text/0040-token-metadata.md

```diff
@@ -0,0 +1,167 @@
+- Feature Name: `token_metadata`
+- Start Date: May 16 2022
+- MCIP PR: [mobilecoinfoundation/mcips#0040](https://github.com/mobilecoinfoundation/mcips/pull/0040)
+- Tracking Issue: (none yet)
+
+# Summary
+[summary]: #summary
+
+Specify a source of truth for metadata about new tokens, and a way for clients to discover this,
+by querying the consensus nodes.
+
+# Motivation
+[motivation]: #motivation
+
+[MCIP #25](https://github.com/mobilecoinfoundation/mcips/pull/0025) and [MCIP #37](https://github.com/mobilecoinfoundation/mcips/pull/0037)
+have specified ways for new tokens to exist on chain. However, for user-facing applications to make effective use of this feature,
+they need more information about a token than just a token id -- they ultimately need to know how they should present balance information
+to the user.
+
+One way to do this is to hard-code all metadata into clients, which might work initially. After this however, this will mean that new tokens
+that are created on chain can't actually be used by users until a client update has been released and the users have updated.
+
+(In the [MCIP #31](https://github.com/mobilecoinfoundation/mcips/pull/0025) implementation PR, we did add a proto enum called `KnownTokenIds`
+which can be shipped with clients, as a stop-gap measure.)
+
+If the clients can dynamically query this information, then a client update is not required for clients to make use of new tokens which are
+created, which significantly reduces the amount of time it takes for a new token to be available to users.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Token metadata includes the following:
+
+* Short (exchange ticker) name of a token, e.g. MOB. (This is not expected to be translated.)
+* Long (english) name of a token, e.g. MobileCoin. (This is expected to be translated, but the client must maintain the translation list.)
+* Display conversion factor, e.g. 10^12. This conversion factor converts from the smallest representable units (`u64`) which appear in the blockchain,
+  to the user-displayable unit. (Note that this is possibly not a power of 10.) `u64` amounts
+  which are obtained from `TxOut`'s on chain should be divided by this number, for the token id, before being displayed to the user.
+* (Optional) URL pointing to an SVG logo of the token. (This is expected to be scalable and appropriate for display on devices of many form factors.)
+* (Optional) URL pointing to an MCIP which proposed the token.
+
+The consensus network will provide an endpoint at which this metadata can be requested for one or more token ids.
```

Comment:
Wanted to note that consensus already provides some information about configured tokens - https://github.com/mobilecoinfoundation/mobilecoin/blob/master/consensus/api/proto/consensus_config.proto#L33
so this proto file could be a good place for this to live. Probably still worth to provide a separate endpoint that only includes whats proposed in this PR since the config endpoint returns a lot of data clients don't necessarily care about.